packages/studio-base/src/panels/Map/getAccuracy.ts

```
@@ -0,0 +1,92 @@
+      // Tilt is degrees from west
+      const eastVariance = covariance[0];
+      const northVariance = covariance[4];
+      return { radii: [Math.sqrt(eastVariance), Math.sqrt(northVariance)], tilt: 0 };
+    }
+    case NavSatFixPositionCovarianceType.COVARIANCE_TYPE_APPROXIMATED:
+    case NavSatFixPositionCovarianceType.COVARIANCE_TYPE_KNOWN: {
+      // Discard altitude
+      const K = covariance;
+      const Klatlon = [
+        [K[0], K[1]],
+        [K[3], K[4]],
+      ];
+
+      // Compute the eigenvalues & vectors of the covariance matrix. They will
+      // be sorted in ascending order, so the largest value is eigenvalues[1]
+      // and the corresponding vector is in the rightmost column. Ellipse radii
+      // are based on the eigenvalues, and orientation on the vector.
+      try {
+        const eigen = eigs(Klatlon) as {
+          vectors: [NumericPair, NumericPair];
+          values: NumericPair;
+        };
+
+        // Eigenvectors are returned in columns
+        const eigenvector = [eigen.vectors[0][1], eigen.vectors[1][1]];
+        const eigenvalues = eigen.values;
+
+        if (
+          !isNumber(eigenvector[0]) ||
+          !isNumber(eigenvector[1]) ||
+          !isNumber(eigenvalues[0]) ||
+          !isNumber(eigenvalues[1])
+        ) {
+          return undefined;
+        }
+
+        // Ellipse `tilt` is defined as number of degrees from the negative x axis
+        const theta = unit(atan2(eigenvector[1], eigenvector[0]), "rad").toNumber("deg");
+        const tilt = -1 * theta;
+
+        const primaryRadius = Math.sqrt(eigenvalues[1]);
+        const secondaryRadius = Math.sqrt(eigenvalues[0]);
+
+        if (isNaN(tilt) || isNaN(primaryRadius) || isNaN(secondaryRadius)) {
+          throw new Error("Unable to calculate accuracy");
+        }
+
+        return {
+          radii: [primaryRadius, secondaryRadius],
+          tilt,
+        };
+      } catch (err) {
+        return undefined;
```

if the error is being discarded, it'd be probably faster to just `return undefined` instead of `throw`ing in the first place